### PR TITLE
Handle `spirv` in addition to `spirv32`/`spirv64` where applicable

### PR DESCRIPTION
**Pull request by alexrp**

### CODE REVIEW

**Comment by alexrp** [comment]
cc @Snektron

File: src/link/SpirV.zig [hunk 1]
```diff
@@ -307,6 +307,7 @@ fn writeMemoryModel(spv: *SpvModule, target: std.Target) !void {
 
     const addressing_model = switch (target.os.tag) {
         .opencl => switch (target.cpu.arch) {
+            .spirv => spec.AddressingModel.Logical,
```
Comment (alexrp) [inline src/link/SpirV.zig:310]: As I'm familiarizing myself with this stuff more, I'm becoming a lot less confident in this particular change. Doesn't *seem* like OpenCL can have logical memory layout?

Comment (alichraghi) [reply src/link/SpirV.zig:310]: [correct](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_Env.html#validation-rules)

Comment (alexrp) [reply src/link/SpirV.zig]: Thanks; removed this part.

**Review by Snektron — APPROVED** [review]

**Review by andrewrk — CHANGES_REQUESTED** [review]
Would you mind rebasing & squashing?

**Comment by alexrp** [comment]
Should be good to go.

**Pull request merged**